src/tracing/esp32_trace/esp32_tracing.cpp

```diff
@@ -155,34 +153,75 @@ void ESP32Backend::TraceCounter(const char * label)
     ::Insights::ESPInsightsCounter::GetInstance(label)->ReportMetrics();
 }
 
+void ESP32Backend::RegisterMetric(const char * key, ValueType type)
+{
+    // Check for the same key will not have two different types.
+    if (mRegisteredMetrics.find(key) != mRegisteredMetrics.end())
+    {
+        if (mRegisteredMetrics[key] != type)
+        {
+            ESP_LOGE("SYS.MTR", "Type mismatch for metric key %s", key);
+            return;
+        }
+        else
+        {
+            return;
```

Comment:
nit: no return after else. This double return could be cleaned up.

Reply:
Done.